code/datums/components/plumbing/reaction_chamber.dm

```diff
@@ -8,13 +8,29 @@
 		return COMPONENT_INCOMPATIBLE
 
 /datum/component/plumbing/reaction_chamber/can_give(amount, reagent, datum/ductnet/net)
+	. = FALSE
+
 	var/obj/machinery/plumbing/reaction_chamber/reaction_chamber = parent
 
 	//cannot give when we outselves are requesting or reacting the reagents
-	if(!reaction_chamber.emptying || reagents.is_reacting)
-		return FALSE
+	if(amount <= 0 || !reagents.total_volume || !reaction_chamber.emptying || reagents.is_reacting)
+		return
+
+	//check to see if we can give catalysts only if they are in excess
+	var/list/datum/reagent/catalysts = reaction_chamber.catalysts
+	for(var/datum/reagent/chemical as anything in reagents.reagent_list)
+		if(reagent && chemical.type != reagent)
+			continue
+
+		//we have the exact amounts so no excess to spare
+		if(chemical.volume <= catalysts[chemical.type] || 0)
```

Comment:
Wait this works?

Reply:
:D

`catalysts[chemical.type]` returns null if its not in the list so we convert that into a number by ORing it with 0

:D

Reply:
I think adding semicolons here here would help with readability
```suggestion
		if(chemical.volume <= (catalysts[chemical.type] || 0))
```